### PR TITLE
[MPIPreferences] Always set `ZES_ENABLE_SYSMAN=1` when dlopening libmpi

### PR DESCRIPTION
**Pull request by giordano**
Diff easier to read if disabling whitespace changes.

Ref: https://github.com/JuliaParallel/MPI.jl/issues/560#issuecomment-1102221341.

### CODE REVIEW

File: lib/MPIPreferences/src/MPIPreferences.jl [hunk 1]
```diff
@@ -157,96 +157,105 @@ function identify_abi(libmpi)
...
-            version = VersionNumber(m.captures[1])
-        end
-    elseif startswith(version_string, "MPIwrapper")
-        impl = "MPIwrapper"
-        # MPIwrapper 2.2.2
-        if (m = match(r"^MPIwrapper Version:\t(\d+.\d+.\d+\w*)", version_string)) !== nothing
-            version = VersionNumber(m.captures[1])
+        @info "MPI implementation" libmpi version_string impl version abi
+
+        if impl == "OpenMPI" && v"4.1.2" ≤ version < v"5"
```
Comment (giordano) [inline lib/MPIPreferences/src/MPIPreferences.jl:248]: I don't know the exact version numbers affected.

Also, I wanted to check the value of `ZES_ENABLE_SYSMAN` here, but it appears that hwloc/openmpi change it after dlopening the library, and trying to access it _before_ dlopening the library also causes a segmentation fault, too :confused:

Comment (vchuravy) [reply lib/MPIPreferences/src/MPIPreferences.jl:248]: Before `dlopen` it should not cause a segmentation, after dlopen and before dlclose maybe?

Comment (giordano) [reply lib/MPIPreferences/src/MPIPreferences.jl:248]: I'll try to build a reproducer later today.

Comment (giordano) [reply lib/MPIPreferences/src/MPIPreferences.jl:248]: BTW, after dlopen and before dlcose it doesn't cause segmentation fault but `ZES_ENABLE_SYSMAN` is always set to 1, so not very useful to look at.  I tried to access it before dlopen with `get(ENV, "ZES_ENABLE_SYSMAN", "")` but I got the segmentation fault upon dlopen :confused:

Comment (simonbyrne) [reply lib/MPIPreferences/src/MPIPreferences.jl:248]: Yup....

**Comment by simonbyrne** [comment]
I don't know what `ZES_ENABLE_SYSMAN=1` does, but should we just set it by default when calling `find_library`?

**Comment by giordano** [comment]
Yeah, good call, I simplified the PR to always wrap dlopening of libmpi with `ZES_ENABLE_SYSMAN=1`.  I guess that's not going to cause any problems for a quick dlopen/dlclose cycle.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**